source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -132,31 +155,26 @@ ParseState Filter::parseHttpHeader(absl::string_view data) {
     protocol_ = "HTTP/2";
     return ParseState::Done;
   } else {
-    const size_t pos = data.find_first_of("\r\n");
-    if (pos != absl::string_view::npos) {
-      const absl::string_view request_line = data.substr(0, pos);
-      const std::vector<absl::string_view> fields =
-          absl::StrSplit(request_line, absl::MaxSplits(' ', 4));
-
-      // Method SP Request-URI SP HTTP-Version
-      if (fields.size() != 3) {
-        ENVOY_LOG(trace, "http inspector: invalid http1x request line");
-        // done(false);
-        return ParseState::Error;
-      }
+    ssize_t rc =
+        http_parser_execute(&parser_, &settings_, data.data() + read_, data.length() - read_);
+    ENVOY_LOG(trace,
+              "http inspector: http_parser parsed {} chars, error code: {}, parser state: {}, "
+              "message complete: {}",
+              rc, HTTP_PARSER_ERRNO(&parser_), parser_.state, message_complete_);
+
+    // Errors in parsing HTTP.
+    if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK && HTTP_PARSER_ERRNO(&parser_) != HPE_PAUSED) {
+      return ParseState::Error;
+    }
 
-      if (http1xMethods().count(fields[0]) == 0 || httpProtocols().count(fields[2]) == 0) {
-        ENVOY_LOG(trace, "http inspector: method: {} or protocol: {} not valid", fields[0],
-                  fields[2]);
-        // done(false);
-        return ParseState::Error;
+    // When the header is parsed, the http_parser will reset the state to s_start_req.
+    if (message_complete_) {
+      if (parser_.http_major == 1 && parser_.http_minor == 1) {
+        protocol_ = Http::Headers::get().ProtocolStrings.Http11String;
+      } else {
+        // Set other HTTP protocols to HTTP/1.0
+        protocol_ = Http::Headers::get().ProtocolStrings.Http10String;
```

Comment:
are you intended to treat HTTP 0.9 as 1.0 ?

Reply:
Yes. This is consistent with HTTP1 codec.
https://github.com/envoyproxy/envoy/blob/c44ac4b0a7b6175dc09f1695f999fd0f7d1141cc/source/common/http/http1/codec_impl.cc#L507-L510